test/mcf/test_ortools_mcf.py

```diff
@@ -163,3 +179,42 @@ def test_unwrap_many_oneworker():
     flows = solver.residues_to_flows_many(residues, cost, worker_count=nworkers)
 
     assert np.ptp(np.ptp(flows, axis=0)) == 0
+
+
+def test_grad_residues():
+    """Test residue computation using points and gradients."""
+    graph, point_data = gen_data_real()
+    solver = spurt.mcf.ORMCFSolver(graph)
+    grads = point_data[solver.edges[:, 1]] - point_data[solver.edges[:, 0]]
+    grads_resid = solver.compute_residues_from_gradients(grads)
+
+    """
+    pts_resid = solver.compute_residue(point_data)
+    print(solver.cycles[0])
+    print(grads_resid[1], pts_resid[1])
+    print(point_data[solver.cycles[0]])
+    edge_to_index = {}
+    for ii, edge in enumerate(solver.edges):
+        edge_to_index[spurt.graph.utils.order_points((edge[0], edge[1]))] = ii
+
+    cyc = solver.cycles[0]
+
+    for ii in range(3):
+        jj = (ii + 1) % 3
+        ind = edge_to_index[spurt.graph.utils.order_points((cyc[ii], cyc[jj]))]
+        print(ind, grads[ind], solver.dual_edges[ind], solver.dual_edge_dir[ind])
+    """
```

Comment:
Happy to take them out. Usually, these are handy when you want to debug something in detail when things go wrong. There doesn't seem to be another location in the repo to include utilities like this.

Reply:
Done

Reply:
>Usually, these are handy when you want to debug something in detail when things go wrong.

Sorry I didn't read through the code carefully since it was commented out. I kind of think its fine to just leave print statements like this uncommented in tests. By default, `pytest` suppresses the output of tests that pass. So these print statements won't affect test output verbosity in the typical case where everything is working fine. You'll only see them when the test fails, which is exactly when you'd want them.

Anyway, either approach is fine with me (removing them or uncommenting them). But keeping the debugging code commented out wouldn't be my preference.